lib/IRGen/IRGen.cpp

```diff
@@ -561,6 +591,45 @@ bool swift::performLLVM(IRGenOptions &Opts, DiagnosticEngine *Diags,
     if (DiagMutex)
       DiagMutex->unlock();
   }
+
+#ifndef NDEBUG
+  if (!OriginalOutputFilename.empty()) {
+    // We're done changing the file; make sure it's saved before we compare.
+    RawOS->close();
+
+    std::string message;
+    if (0 != llvm::DiffFilesWithTolerance(OutputFilename,
+                                          OriginalOutputFilename,
+                                          /*tolerances=*/0, 0, &message)) {
+      if (DiagMutex)
+        DiagMutex->lock();
```

Comment:
I don't think you need to take the mutex here (or below); you're just going to crash anyway, and the temporary file doesn't affect other threads.

Reply:
The mutex is to keep several threads from writing messages at the same time (or at least that's my reading of why it's done elsewhere in the file), but I think I'll simplify the output a bit anyway.